File: instrumentation/okhttp-3.6.0/src/main/java/com/nr/agent/instrumentation/okhttp36/InboundWrapper.java
/*
 *
 *  * Copyright 2020 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.nr.agent.instrumentation.okhttp36;

import com.newrelic.api.agent.HeaderType;
import com.newrelic.api.agent.ExtendedInboundHeaders;
import okhttp3.Response;

import java.util.List;

public class InboundWrapper extends ExtendedInboundHeaders {

    private final Response response;

    public InboundWrapper(Response response) {
        this.response = response;
    }

    @Override
    public HeaderType getHeaderType() {
        return HeaderType.HTTP;
    }

    @Override
    public String getHeader(String name) {
        return response.header(name);
    }

    @Override
    public List<String> getHeaders(String name) {
        return response.headers(name);
    }

}

File: instrumentation/spray-http-1.3.1/src/main/scala/com/agent/instrumentation/spray/InboundWrapper.scala
/*
 *
 *  * Copyright 2020 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.agent.instrumentation.spray

import java.util

import com.newrelic.api.agent.{ExtendedInboundHeaders, HeaderType}
import spray.http.HttpRequest

import scala.collection.JavaConversions

class InboundWrapper(request: HttpRequest) extends ExtendedInboundHeaders {

  def getHeaderType: HeaderType = {
    HeaderType.HTTP
  }

  def getHeader(name: String): String = {
    request.headers.find(header => header.is(name.toLowerCase)).map(header => header.value).orNull
  }

  override def getHeaders(name: String): util.List[String] = {
    val headers = request.headers.filter(header => header.is(name.toLowerCase)).map(header => header.value)
    if (headers.isEmpty) {
      return null
    }
    JavaConversions.seqAsJavaList(headers)
  }

}

